File: app/src/main/java/com/jgasteiz/comics_android/models/Series.java
package com.jgasteiz.comics_android.models;

import android.util.Log;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class Series implements Serializable {

    private static final String LOG_TAG = Series.class.getSimpleName();

    private int id;
    private String title;
    private String author;
    private String year;

    public Series() {}

    public Series(JSONObject jsonObject) {
        try {
            this.id = (int) jsonObject.get("pk");
            this.title = (String) jsonObject.get("title");
            this.author = (String) jsonObject.get("author");
            this.year = (String) jsonObject.get("year");
        } catch (JSONException e) {
            Log.e(LOG_TAG, e.getMessage());
        }
    }

    public static String getLogTag() {
        return LOG_TAG;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getYear() {
        return year;
    }

    public void setYear(String year) {
        this.year = year;
    }
}

File: app/src/main/java/com/jgasteiz/comics_android/models/Series.kt
package com.jgasteiz.comics_android.models

import android.util.Log
import org.json.JSONException
import org.json.JSONObject

import java.io.Serializable

class Series : Serializable {

    val logTag = "Series"

    var id: Int = 0
    var title: String? = null
    var author: String? = null
    var year: String? = null

    constructor(id: Int, title: String, author: String, year: String) {
        this.id = id
        this.title = title
        this.author = author
        this.year = year
    }

    constructor(jsonObject: JSONObject) {
        try {
            this.id = jsonObject.get("pk") as Int
            this.title = jsonObject.get("title") as String
            this.author = jsonObject.get("author") as String
            this.year = jsonObject.get("year") as String
        } catch (e: JSONException) {
            Log.e(logTag, e.message)
        }

    }
}
